2025/lab/Keys/Week8_Binary_Outcomes.Rmd

```diff
@@ -0,0 +1,406 @@
+      race == 3 & hispanic == 1 ~ "Other",
+      ),
+    race4 = factor(race, levels = c("White", "Black", "Hispanic", "Other")),
+    weight =  wtssnrps 
+    )  %>%   
+  filter(ballot != 1) %>% 
+  select(-wtssnrps , -hispanic, -race, -polviews, -grassnv, -ballot)
+
+
+```
+
+Let's take a look at our two new variables. 1) `legal_grass`, which asks whether marijuana should be legalized; 2) `conservatism`, which is a 1 to 7 scale of ideology, with 1 as extremely liberal and 7 as extremely conservative. Let's double check the coding below. 
+
+## Missingness Analysis
+
+```{r miss-upset}
+gg_miss_upset(gss)
+```
+While we can (and probably should) do any of the methods from last time to impute missing values, we're just going to use listwise deletion to keep it simple. 
+
+```{r}
+gss <- gss %>% drop_na()
+```
+
+
+## Descriptives
+As always, let's look at our descriptive statistics. 
+
+```{r desc-tables}
+datasummary_skim(gss, type = "numeric",
+                 title = "Sample Descriptive Statistics: Continuous Variables",
+                 notes = "Data: 2022 General Social Survey")
+
+datasummary_skim(gss, type = "categorical",
+                 title = "Sample Descriptive Statistics: Categorical Variables",
+                 notes = "Data: 2022 General Social Survey")
+
+```
+
+We can also create a "balance table" to compares demographics on each side of the issue.
+
+The format is `datasummary_balance(~VAR, data = DATA, ... )`, where `VAR` is the variable we want to analyze the sample by. Note the function only provides differences and standard errors of those differences for continuous variables and only when we split by two groups.  
+
+```{r baltable}
+datasummary_balance(~legal_grass, data = gss,
+                    title = "Balance Table: Demographics by Views on Marijuana Legalization",
+                    notes = "Data: 2022 General Social Survey (Unweighted)")
+```
+
+Looking at the column "Diff. in Means," we can see that people who favor marijuana legalization tend to be younger by about 8 years, are more liberal by close to one step, and have about one-half more years of education. We also see that people who support legalization are more likely to be Democrats and male. 
+
+\newpage 
+
+# Linear Probability Model
+
+Let's start off by creating a linear probability model for the likelihood somebody favors marijuana legalization. 
+
+```{r lin-prob}
+linprob1 <- lm(legal_grass ~ conservatism + age + educ + race4 + sex,
+               data = gss, weights = weight)
+modelsummary(linprob1, estimate = "{estimate} ({std.error})", 
+             statistic = NULL, 
+             gof_omit = "F|RMSE|Log|IC",
+             output = "huxtable")
+```
+
+
+## Interpretation
+
+We can interpret this model as the effect of various factors on the probability a person is in favor of legalizing marijuana. So each additional step of conservatism decreases this probability by .098 and each additional year of age decreases it by .005. 
```

Comment:
Maybe it is a pain to change. I'll let you decide, but I'm not a fan of treating a categorical like this as linear.